src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -2539,7 +2537,7 @@ public XmlDataDocument() : base(new XmlDataImplementation())
         {
             Init();
             AttachDataSet(new DataSet());
-            _dataSet.EnforceConstraints = false;
+            _dataSet!.EnforceConstraints = false;
```

Comment:
(nit) if `AttachDataSet` was annotated with `[MemberNotNull(nameof(_dataSet))]`, this `!` wouldn't be needed.

Reply:
Good call! Fixed.